src/ml_gym/frontend/dashboard/src/features/experimentConfig/experimentConfigSlice.tsx

```diff
@@ -0,0 +1,42 @@
+import { createSlice } from '@reduxjs/toolkit'
+import { ExperimentConfigMessageCollectionType, ExperimentConfigType} from '../../app/datatypes'
+import type { RootState } from '../../app/store';
+
+const initialState = {
+    messages: [],
+    experiment_id_to_message_index: {}
+} as ExperimentConfigMessageCollectionType
+
+
+const experimentConfigSlice = createSlice({
+    name: 'experimentConfig',
+    initialState,
+    reducers: {
+        experimentConfigAdded(state, action) {
+            const experiment_id: number = action.payload.data.payload.experiment_id
+            if (experiment_id in state.experiment_id_to_message_index) {
+                const row_id = state.experiment_id_to_message_index[experiment_id]
+                // prevents possible race condition
+                if (state.messages[row_id].event_id < action.payload.event_id) {
+                    state.messages[row_id] = action.payload
+                }
+            } else {
+                state.messages.push(action.payload)
+                state.experiment_id_to_message_index[experiment_id] = state.messages.length - 1
```

Comment:
Codacy has a fix for the issue: [Replace `················state.experiment_id_to_message_index[experiment_id]·=·state.messages.length·-·1` with `········state.experiment_id_to_message_index[experiment_id]·=⏎··········state.messages.length·-·1;`](https://app.codacy.com/gh/mlgym/mlgym/pullRequest?prid=10020887)
```suggestion
        state.experiment_id_to_message_index[experiment_id] =
```